internal/core/services/scheduler_manager/scheduler_manager.go

```diff
@@ -123,72 +146,48 @@ func (s *SchedulerManager) RemoveRooms(ctx context.Context, schedulerName string
 	return op, nil
 }
 
-// UpdateSchedulerConfig receives the configuration of a scheduler, generate a new
-// version and update it on the scheduler's storage. It returns if the update
-// was a major update or not.
-// Modifies the provided scheduler Spec.Version and RollbackVersion.
-//
-// TODO(gabrielcorado): should we update if no changes were made?
-func (s *SchedulerManager) UpdateSchedulerConfig(ctx context.Context, scheduler *entities.Scheduler) (bool, error) {
-	err := scheduler.Validate()
-	if err != nil {
-		return false, err
-	}
-
+func (s *SchedulerManager) EnqueueNewSchedulerVersionOperation(ctx context.Context, scheduler *entities.Scheduler) (*operation.Operation, error) {
 	currentScheduler, err := s.schedulerStorage.GetScheduler(ctx, scheduler.Name)
-	if err != nil {
-		return false, fmt.Errorf("error fetching scheduler: %w", err)
+	if err != nil || currentScheduler == nil {
```

Comment:
Why is this condition "currentScheduler == nil" necessary? The GetScheduler method returns nil, err in case the query on database returns 0 rows...

Reply:
This nil check is unnecessary